File: solutions/day_1/contains_duplicate.dart
class Solution {
  bool containsDuplicate(List<int> nums) {
    Set<int> checkedVar = Set();

    for (int num in nums) {
      //"!" very important, if you don't use it, it will return false even if there is a duplicate
      if (!checkedVar.contains(num)) {
        checkedVar.add(num);
      } else
        return true;
    }
    return false;
  }
}

File: contains-duplicate/contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]):
        left, right = 0, 1

        nums = sorted(nums)
        nums_len = len(nums)

        if nums_len == 1 or nums_len == 0:
            return False

        while right < nums_len:
            if nums[left] == nums[right]:
                return True

            left += 1
            right += 1

        return False
